community/CONTRIBUTING.md

```diff
@@ -5,6 +5,192 @@ Thanks for considering contributing to OPEA project. The contribution process is
+
+- Follow the folder structure in the ChatQA example below:
+
+```
+├── assets
+├── benchmark     # optional
+├── chatqna.py    # Main application definition (microservices, megaservice, gateway). 
+├── chatqna.yaml  # starting v1.0 used to generate manifests for k8s w orchestrator_with_yaml
+├── docker_compose
+│   ├── intel
+│   │   ├── cpu
+│   │   │   └── xeon
+│   │   │       ├── compose.yaml
+│   │   │       ├── README.md 
+│   │   │       └── set_env.sh  #export env variables
+│   │   └── hpu
+│   │       └── gaudi
+│   │           ├── compose.yaml
+│   │           ├── how_to_validate_service.md    #optional
+│   │           ├── README.md
+│   │           └── set_env.sh
+│   └── nvidia
+│       └── gpu   
+│           ├── compose.yaml
+│           ├── README.md
+│           └── set_env.sh
+├── Dockerfile
+├── docker_image_build
+│   └── build.yaml
+├── kubernetes
+│   ├── intel
+│   │   ├── cpu
+│   │   │   └── xeon
+│   │   │       ├── gmc
+│   │   │       │   └── chatQnA_xeon.yaml
+│   │   │       └── manifest
+│   │   │           └── chatqna.yaml     
+│   │   └── hpu
+│   │       └── gaudi
+│   │           ├── gmc
+│   │           │   └── chatQnA_gaudi.yaml
+│   │           └── manifest
+│   │               └── chatqna.yaml
+│   ├── amd
+│   │   ├── cpu      
+│   │   │   ├── gmc
+│   │   │   └── manifest
+│   │   └── gpu
+│   │       ├── gmc
+│   │       └── manifest
+│   ├── README_gmc.md  # K8s quickstar
+│   └── README.md      # quickstart
+├── README.md
+├── tests
+│   ├── test_compose_on_gaudi.sh  #could be more tests for different flavors of the app
+│   ├── test_gmc_on_gaudi.sh
+│   ├── test_manifest_on_gaudi.sh
+└── ui
+
+```
+- **File Descriptions**:
+  - `chatqna.py`: application definition using microservice, megaservice and gateway. There could be multiple .py in the folder based on slight modification of the example application.
+  - `docker_build_image/build.yaml`: builds necessary images pointing to the Dockerfiles in the GenAIComp repository. 
+  - `docker_compose/vendor/device/compose.yaml`: defines pipeline for  docker compose deployment. For selectng docker image name please follow the naming convention: 
+      - Docker Image: `opea/[example name]-[feature name]:latest` all lower case (i,e: opea/chatqna, opea/codegen-react-ui)
+  - `kubernetes/vendor/device/manifests/chatqna.yaml`: used for K8s deployemnt
+  - `kubernetes/vendor/device/gmc/chatqna.yaml`: (optional) used for deployment with GMC
```

Comment:
maybe we need a brief introduce on what's the GMC at the first time seeing it.

Reply:
added intro at the beginning of the section